File: asmTest/src/main/java/com/test/MainActivity.java
package com.test;

import android.os.Bundle;
import android.util.Log;

import androidx.annotation.Nullable;
import androidx.fragment.app.FragmentActivity;


public class MainActivity extends FragmentActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.e("TAG", "com/test/MainActivity------->onCreate");
        setContentView(R.layout.activity_main);
        MainActivity.test(10, "11", 12, "13");
    }

    public static int test(int a, String b, int c, String d) {
        int z = 1;
        Runnable runnable = new Runnable() {
            @Override
            public void run() {

            }
        };
        Thread thread = new Thread(runnable);
        thread.setName("original");
        thread.start();
        return 0;
    }
}

File: Dokit/dokit_test/src/main/java/com/dokit/test/MainActivity.kt
package com.dokit.test

import android.os.Bundle
import android.util.Log
import androidx.fragment.app.FragmentActivity
import com.test.kotlin.block.BlockTest
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : FragmentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        blockTest.setOnClickListener {
            val blockTest = BlockTest()
            blockTest.test1 {
                Log.v("AndroidTest", "MainActivity===>test1")
            }

            blockTest.test2 {
                Log.v("AndroidTest", "MainActivity===>test1")
            }

            blockTest.test3 {
                Log.v("AndroidTest", "MainActivity===>test2")
                "1"
            }

            blockTest.test4 { x, y ->
                Log.v("AndroidTest", "MainActivity===>test3===>x = $x, y = $y")
                x + y
            }

            blockTest.test5 { x, y ->
                Log.v("AndroidTest", "MainActivity===>test4===>x = $x, y = $y")
            }
        }
    }
}